draft-ietf-httpbis-http2bis.xml

```diff
@@ -606,6 +600,49 @@ HTTP Frame {
           connection with a <xref target="ConnectionErrorHandler">connection error</xref> of type
           <xref target="COMPRESSION_ERROR" format="none">COMPRESSION_ERROR</xref> if it does not decompress a field block.
         </t>
+        <t>
+          A decoding error in a field block MUST be treated as a <xref
+          target="ConnectionErrorHandler">connection error</xref> of type <xref
+          target="COMPRESSION_ERROR" format="none">COMPRESSION_ERROR</xref>.
+        </t>
+
+        <section anchor="dynamic-table">
+          <name>Compression State</name>
+          <t>
+            Field compression is stateful.  One compression context and one decompression context
+            are used for the entire connection.  <xref target="COMPRESSION" section="4"/> defines
+            the dynamic table, which is the primary state that is used for field compression.
+          </t>
+          <t>
+            The dynamic table has a maximum size that is set by a decoder using the
+            SETTINGS_HEADER_TABLE_SIZE setting; see <xref target="SettingValues"/>.  Any change to
+            the maximum value set by the decoder takes effect when the encoder <xref
+            target="SettingsSync">acknowledges settings</xref>.  The encoder can set the dynamic
+            table to any size up to the maximum value set by the decoder.  The encoder declares the
+            size of the dynamic table with a Dynamic Table Size Update instruction (<xref
+            target="COMPRESSION" section="6.3"/>).  The encoder at both client and server is
+            initialized with a dynamic table size of 4,096 bytes, the initial value of the
+            SETTINGS_HEADER_TABLE_SIZE setting.
+          </t>
+          <t>
+            If a change to SETTINGS_HEADER_TABLE_SIZE reduces the maximum below the current size of
+            the dynamic table, the encoder MUST start the next field block with a Dynamic Table Size
+            Update instruction that sets the dynamic table to a size that is less than or equal to
+            the reduced maximum; see <xref target="COMPRESSION" section="4.2"/>.  A decoder MUST
```

Comment:
This is incorrect. SETTINGS_MAX_HEADER_LIST_SIZE is merely advisory. Encoders are allowed to ignore the the SETTINGS value ([section 10.5.1](https://httpwg.org/http2-spec/draft-ietf-httpbis-http2bis.html#name-limits-on-field-block-size)).

Reply:
Warning @kazuho, the part in section 10.5.1 is about uncompressed fields, it only says that it's possible to send fields that are larger than the table size, and that this settings may or may not be used as a hint for such a size limit. I personally don't establish any such relation between maximum field size and headers table size, since HPACK clearly states that trying to store a larger field simply flushes the table, and I think that most implementations will use small tables even is they support large headers. I think the text really focuses on a hint that if an implementation advertises a large table, it's likely that it will also support large fields.